finetuner/tuner/pytorch/__init__.py

```diff
@@ -222,7 +224,24 @@ def save(self, *args, **kwargs):
         :param args: Arguments to pass to ``torch.save`` function
         :param kwargs: Keyword arguments to pass to ``torch.save`` function
         """
-        torch.save(self.embed_model.state_dict(), *args, **kwargs)
+
+        state = {
+            'epoch': kwargs.pop('epoch', 0),
+            'state_dict': self.embed_model.state_dict(),
+            'best': kwargs.pop('best', False),
+            'optimizer': self._optimizer.state_dict(),
+            'monitor': kwargs.pop('monitor', 'train_loss'),
+        }
+
+        torch.save(state, *args, **kwargs)
+
+    def load(self, fp):
+        """Loads the embedding model, optimizer and updates the state epoch."""
+
+        checkpoint = torch.load(fp)
+        self._embed_model.load_state_dict(checkpoint['state_dict'])
+        self._optimizer.load_state_dict(checkpoint['optimizer'])
+        self.state.epoch = checkpoint['epoch']
```

Comment:
Same here - leave loading/saving of trainer state to the checkpoint, do not couple it with the trainer itself